implants/lib/eldritch/src/file/find_impl.rs

```diff
@@ -0,0 +1,80 @@
+use std::{path::{Path, PathBuf}, time::{SystemTime, UNIX_EPOCH}};
+use anyhow::{anyhow, Result};
+#[cfg(unix)]
+use std::os::unix::fs::PermissionsExt;
+
+fn check_path(path: &PathBuf,  name: Option<String>, file_type: Option<String>, permissions: Option<u64>, modified_time: Option<u64>, create_time: Option<u64>) -> Result<bool> {
+    if let Some(name) = name {
+        if !path.file_name().unwrap().to_str().unwrap().contains(&name) {
+            return Ok(false);
+        }
+    }
+    if let Some(file_type) = file_type {
+        if !path.is_file() && file_type == "file" {
+            return Ok(false);
+        }
+        if !path.is_dir() && file_type == "dir" {
+            return Ok(false);
+        }
+    }
+    if let Some(permissions) = permissions {
+        let metadata = path.metadata()?.permissions();
+        #[cfg(unix)]
+        {
+            if metadata.mode() != (permissions as u32) {
+                return Ok(false);
+            }
+        }
+        #[cfg(windows)]
+        {
+            if permissions == 0 && metadata.readonly() {
```

Comment:
Could be cleaner to switch to a match.

Reply:
Problem is that for unix I'm making calls to a permissions trait in `PermissionsExt`, so if we don't do this then the Windows version will try and compile with the use of the unix-only trait and error

Reply:
Couldn't it still be bound in the `#[cfg(windows)]` ?
I'm thinking
```rust
#[cfg(windows)]
{
    match permissions {
        0 => {
            if metadata.readonly() { return Ok(false) }
        }
       1 => {
            if !metadata.readonly() { return Ok(false) }
       }
       _ => {
          return anyhow::anyhow!("Windows only supports readonly (0) or not-readonly (1)")
      }
    }
}
```